authentication/auth_method_azure_cli_token.go

```diff
@@ -172,6 +190,50 @@ func (a azureCliTokenAuth) validate() error {
 	return err.ErrorOrNil()
 }
 
+func (a azureCliTokenAuth) checkAzVersion() error {
+	// Azure CLI v2.0.79 is the earliest version to have a `version` command
+	var minimumVersion string
+	if a.profile.tenantOnly {
+		// v2.0.81 introduced the `--tenant` option to the `account get-access-token` subcommand
+		minimumVersion = "2.0.81"
+	} else {
+		minimumVersion = "2.0.79"
+	}
+
+	var cliVersion *struct {
+		AzureCli          *string      `json:"azure-cli"`
+		AzureCliCore      *string      `json:"azure-cli-core"`
+		AzureCliTelemetry *string      `json:"azure-cli-telemetry"`
+		Extensions        *interface{} `json:"extensions"`
+	}
+	err := jsonUnmarshalAzCmd(&cliVersion, "version", "-o=json")
+	if err != nil {
+		return fmt.Errorf("Please ensure you have installed Azure CLI version %s or newer. Error parsing json result from the Azure CLI: %v.", minimumVersion, err)
+	}
+
+	if cliVersion.AzureCli == nil {
+		return fmt.Errorf("Could not detect Azure CLI version. Please ensure you have installed Azure CLI version %s or newer.", minimumVersion)
+	}
+
+	actual, err := version.NewVersion(*cliVersion.AzureCli)
+	if err != nil {
+		return fmt.Errorf("Could not parse detected Azure CLI version: %+v", err)
+	}
+
+	supported, err := version.NewVersion(minimumVersion)
+	if err != nil {
+		return fmt.Errorf("Could not parse supported Azure CLI version: %+v", err)
+	}
+
+	if supported.LessThanOrEqual(actual) {
```

Comment:
maybe overkill, do we want to check it's less than 3.0 too?

Reply:
Good call, have implemented :+1: